src/databricks/labs/ucx/migration/azure_credentials.py

```diff
@@ -0,0 +1,285 @@
+                    ServicePrincipalMigrationInfo.from_storage_permission_mapping(
+                        sp, azure_sp_info_with_client_secret[sp.client_id]
+                    )
+                )
+        return sp_list_with_secret
+
+    def _print_action_plan(self, sp_list_with_secret):
+        # print action plan to console for customer to review.
+        for sp in sp_list_with_secret:
+            print(
+                f"Service Principal name: {sp.principal}, application_id: {sp.client_id}, privilege {sp.privilege} on location {sp.prefix}"
+            )
+
+    def _generate_migration_list(self):
+        """
+        Create the list of SP that need to be migrated, output an action plan as a csv file for users to confirm
+        :return:
+        """
+        # load sp list from azure_storage_account_info.csv
+        sp_list = self._azure_resource_permissions.load()
+        # list existed storage credentials
+        sc_set = self._list_storage_credentials()
+        # check if the sp is already used in UC storage credential
+        filtered_sp_list = self._check_sp_in_storage_credentials(sp_list, sc_set)
+        # fetch sp client_secret if any
+        sp_list_with_secret = self._fetch_client_secret(filtered_sp_list)
+        self._final_sp_list = sp_list_with_secret
+        # output the action plan for customer to confirm
+        self._print_action_plan(sp_list_with_secret)
+        return
+
+    def _create_storage_credential(self, sp: ServicePrincipalMigrationInfo):
+        # prepare the storage credential properties
+        name = sp.principal
+        azure_service_principal = AzureServicePrincipal(
+            directory_id=sp.directory_id, application_id=sp.client_id, client_secret=sp.client_secret
+        )
+        comment = f"Created by UCX during migration to UC using Azure Service Principal: {sp.principal}"
+        read_only = False
+        if sp.privilege == Privilege.READ_FILES:
+            read_only = True
+        # create the storage credential
+        storage_credential = self._ws.storage_credentials.create(
+            name=name, azure_service_principal=azure_service_principal, comment=comment, read_only=read_only
+        )
+
+        validation_result = self._validate_storage_credential(storage_credential, sp.prefix)
+        yield validation_result
+
+    def _validate_storage_credential(self, storage_credential, location) -> StorageCredentialValidationResult:
+        validation = self._ws.storage_credentials.validate(
+            storage_credential_name=storage_credential.name, url=location
+        )
+        return StorageCredentialValidationResult.from_storage_credential_validation(storage_credential, validation)
+
+    def execute_migration(self, prompts: Prompts):
+        if not self._ws.config.is_azure:
+            logger.error("Workspace is not on azure, please run this command on azure databricks workspaces.")
+            return
+
+        csv_confirmed = prompts.confirm(
```

Comment:
All the prompts should be part of the for_cli method.

Reply:
fixed